contents/blog/the-posthog-array-1-41-0.md

```diff
@@ -0,0 +1,146 @@
+---
+date: 2022-10-31
+title: "Array 1.41.0: Enable hedgehog mode"
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-array-blog.png
+featuredImageType: full
+---
+
+Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send every two weeks!
+
+> Running a self-hosted instance? Check out our [guide to upgrading PostHog](/docs/runbook/upgrading-posthog).
+
+> **IMPORTANT NOTE FOR SELF-HOSTED USERS:** Self-hosted users should check `alias` usage before [upgrading PostHog](/docs/runbook/upgrading-posthog) due to changes involving persons on events. For example, assuming `email` is used as the identified user id, then: 
+
+```
+identify(email)                                   # in the frontend
+alias(backend_unique_id, email)      # in the backend - THIS WILL NOT WORK
+alias(email, backend_unique_id2)    # in the backend - the right order
+```
+After updating to 1.41.0, check [ingestion warnings](#new-ingestion-warnings) and run async migrations 0005, 0006 and 0007.
+
+## PostHog 1.41.0 release notes
+
+**Release highlights:**
+
+- New: Persons on events on by default
+- New: Count of events per user
+- New: Text cards on dashboards
+- New: App metrics
+- New: Ingestion warnings
+- New: Change your email yourself
+- New: Hedgehog mode
+- Improved: Recordings interface
+- One more thing: Site apps 
+
+### New: Persons on events on by default
+
+We used to store events in one table and persons in another table. That meant that, once you reached billion event scale, any query which touched person properties would time out. But, no longer! After running [an extensive beta since update 1.39.0](/blog/the-posthog-array-1-39-0#beta-improving-query-performance-by-combining-persons-and-events), we've now moved person data onto the events themselves. This change also applies to groups, adding group properties and aggregation to the same table. 
+
+You won’t see any UI changes as a result of this change — persons will still have their own Persons & Groups section on the sidebar, for example — but you will notice results are a lot (up to 400%!) faster for any queries involving persons, groups or events. This is a massive change, so be sure to [read the full announcement for more info](/blog/persons-on-events).
+
+> **Note:** As a result of putting persons on events we now have to be strict about identified and unidentified users. Now, only anonymous users can be aliased into other users - meaning that once you call `identify` on a user, you can no longer alias the user into another identified user. Because of that, we recommend double checking your `identify` and `alias` calls, and following advice above [how to upgrade a self-hosted instance to 1.41.0](/blog/the-posthog-array-1-41-0). 
```

Comment:
for alias changes check out this email that we sent to the users https://docs.google.com/document/d/1qbd52Mmf8DgOPbFB4bs2vAHNi1gSPNw2kGxPD4Khewk/edit & my outdated comment.

Reply:
@joethreepwood  lets just make it be short and simple here:

```suggestion
> **Note:** As a result of adding persons on events and changing how queries work we now have to be more strict about merging users. Because of that, we recommend double checking your `alias` calls, and following advice above [how to upgrade a self-hosted instance to 1.41.0](/blog/the-posthog-array-1-41-0). 
```